pyaedt/edb_core/materials.py

```diff
@@ -856,116 +845,133 @@ def add_material_from_aedt(self, material_name):
-                except (KeyError, TypeError):
-                    pass
-                try:
-                    new_material.mass_density = float(material["mass_density"])
-                except (KeyError, TypeError):
-                    pass
-                try:
-                    new_material.permeability = float(material["permeability"])
-                except (KeyError, TypeError):
-                    pass
-                try:
-                    new_material.loss_tangent = float(material["dielectric_loss_tangent"])
-                except (KeyError, TypeError):
-                    pass
-                try:
-                    new_material.specific_heat = float(material["specific_heat"])
-                except (KeyError, TypeError):
-                    pass
-                try:
-                    new_material.thermal_expansion_coefficient = float(material["thermal_expansion_coeffcient"])
-                except (KeyError, TypeError):
-                    pass
+                properties = [
+                    "permittivity",
+                    "conductivity",
+                    "mass_density",
+                    "permeability",
+                    "specific_heat",
+                    "thermal_expansion_coefficient",
+                ]
+                for mat_prop_name, mat_prop_value in material.items():
+                    if mat_prop_name in properties:
+                        setattr(new_material, mat_prop_name, mat_prop_value)
+                    if mat_prop_name == "tangent_delta":
+                        new_material.loss_tangent = mat_prop_value
         return True
 
+    @staticmethod
     @pyaedt_function_handler()
-    def _read_materials(self, mat_file=None):
-        def get_mat_list(file_name, mats):
-            from pyaedt.generic.LoadAEDTFile import load_entire_aedt_file
-
-            mread = load_entire_aedt_file(file_name)
-            for mat, mdict in mread.items():
-                if mat != "$base_index$":
-                    try:
-                        mats[mat] = mdict["MaterialDef"][mat]
-                    except KeyError:
-                        mats[mat] = mdict
-
-        if mat_file and os.path.exists(mat_file):
-            materials = {}
-            get_mat_list(mat_file, materials)
-            return materials
-
-        amat_sys = [
-            os.path.join(dirpath, filename)
-            for dirpath, _, filenames in os.walk(self.syslib)
-            for filename in filenames
-            if fnmatch.fnmatch(filename, "*.amat")
+    def read_materials(mat_file):
+        """Read materials from an AMAT file.
+
+        Parameters
+        ----------
+        filename : str
+            Name of the AMAT file.
+
+        Returns
+        -------
+        dict
+            {material name: dict of material property to value}.
+        """
+
+        def get_line_float_value(line):
+            try:
+                return float(re.split(",|=", line)[-1].strip(")'"))
```

Comment:
Maybe here, just add a comment to explain what you are doing.
It would make futur maintenance easier.